### PR TITLE
[FIX] pos_restaurant: release table after order is completed

### PR DESCRIPTION
**Pull request by jipr-odoo**
Before this commit:
====================
The table remained occupied after the order was completed (payment done)

After this commit:
===================
Tables are now released once the order is fully completed, ensuring they are available for future use.

Task- 4228671

### CODE REVIEW

File: addons/pos_restaurant/static/src/overrides/models/pos_store.js [hunk 1]
```diff
@@ -319,7 +319,10 @@ patch(PosStore.prototype, {
         );
     },
     tableHasOrders(table) {
-        return Boolean(table.getOrder());
+        return (
+            this.models["pos.order"].filter((o) => o.table_id?.id === table.id && !o.finalized)
+                .length > 0
+        );
```
Comment (paso-odoo) [inline addons/pos_restaurant/static/src/overrides/models/pos_store.js:325]: I would like to apply same logic in `getOrder`

Comment (jipr-odoo) [reply addons/pos_restaurant/static/src/overrides/models/pos_store.js:325]: Sure, now it's 18.0, so we shouldn't remove any method.

File: addons/pos_restaurant/static/src/app/models/restaurant_table.js [hunk 2]
```diff
@@ -71,7 +71,11 @@ export class RestaurantTable extends Base {
...
     getOrder() {
-        return this.parent_id?.getOrder?.() || this["<-pos.order.table_id"][0];
+        return (
+            this.parent_id?.getOrder?.() ||
+            this.models["pos.order"].filter((o) => o.table_id?.id === this.id && !o.finalized)
+                .length > 0
+        );
     }
     setPositionAsIfLinked(parent, side) {
         // console.log("132")
```
Comment (paso-odoo) [inline addons/pos_restaurant/static/src/app/models/restaurant_table.js:81]: remove this too

File: addons/pos_restaurant/static/src/app/models/restaurant_table.js [hunk 3]
```diff
@@ -71,10 +71,13 @@ export class RestaurantTable extends Base {
         );
     }
     getOrder() {
-        return this.parent_id?.getOrder?.() || this["<-pos.order.table_id"][0];
+        return (
+            this.parent_id?.getOrder?.() ||
+            this.models["pos.order"].filter((o) => o.table_id?.id === this.id && !o.finalized)
+                .length > 0
+        );
```
Comment (caburj) [inline addons/pos_restaurant/static/src/app/models/restaurant_table.js:78]:
```suggestion
        return this.parent_id?.getOrder?.() || this["<-pos.order.table_id"].find((o) => !o.finalized);
```

can you try this?

File: addons/pos_restaurant/static/src/overrides/models/pos_store.js [hunk 4]
```diff
@@ -319,7 +319,7 @@ patch(PosStore.prototype, {
         );
     },
     tableHasOrders(table) {
-        return Boolean(table.getOrder());
+        return table.getOrder();
```
Comment (caburj) [inline addons/pos_restaurant/static/src/overrides/models/pos_store.js:322]:
```suggestion
        return Boolean(table.getOrder());
```

Restore this diff. I think only the change is `getOrder` is needed.

File: addons/pos_restaurant/static/src/app/models/restaurant_table.js [hunk 5]
```diff
@@ -71,10 +71,13 @@ export class RestaurantTable extends Base {
...
     }
     setPositionAsIfLinked(parent, side) {
-        // console.log("132")
```
Comment (caburj) [inline addons/pos_restaurant/static/src/app/models/restaurant_table.js:77]: :exploding_head:

File: addons/pos_restaurant/static/src/app/models/restaurant_table.js (hunk 3 above)
Comment (jipr-odoo) [reply addons/pos_restaurant/static/src/app/models/restaurant_table.js]: Great, I've made the changes. Please review. Thank you!

**Pull request closed without merging**